presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java

```diff
@@ -78,8 +78,14 @@ public SessionMatchSpec(
             }
         }
 
-        if (resourceGroupRegex.isPresent() && !resourceGroupRegex.get().matcher(context.getResourceGroupId().toString()).matches()) {
-            return ImmutableMap.of();
+        if (resourceGroupRegex.isPresent()) {
+            String resourceGroupId = "";
```

Comment:
nit: `context.getResourceGroupId().orElse("")`

Reply:
You meant `context.getResourceGroupId().orElse(new ResourceGroupId(""))` ?  This will fail as we cannot use empty string to construct `ResourceGroupId`